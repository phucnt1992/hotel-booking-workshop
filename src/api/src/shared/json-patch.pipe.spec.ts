import { JsonPatchPipe } from './json-patch.pipe';

describe('JsonPatchPipe', () => {
  it('should be defined', () => {
    expect(new JsonPatchPipe()).toBeDefined();
  });

  it('should be invalid with invalid patch value', () => {
    const patches = [
      { wrong: 'replace', path: '/user/firstName', value: 'Albert' },
    ];

    expect(() =>
      new JsonPatchPipe().transform(patches),
    ).toThrowErrorMatchingSnapshot();
  });

  it('should be valid with valid patche value', () => {
    const patches = [
      { op: 'replace', path: '/user/firstName', value: 'Albert' },
    ];

    expect(new JsonPatchPipe().transform(patches)).toEqual(patches);
  });
});
