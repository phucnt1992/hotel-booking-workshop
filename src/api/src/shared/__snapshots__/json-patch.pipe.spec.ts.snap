// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JsonPatchPipe should be invalid with invalid patch value 1`] = `
Object {
  "error": "Bad Request",
  "message": "Operation \`op\` property is not one of operations defined in RFC-6902
name: OPERATION_OP_INVALID
index: 0
operation: {
  \\"wrong\\": \\"replace\\",
  \\"path\\": \\"/user/firstName\\",
  \\"value\\": \\"Albert\\"
}",
  "statusCode": 400,
}
`;
